### PR TITLE
Validates if `memosOrder.length` equals to `memosAmount`

### PR DESCRIPTION
**Pull request by ggirotto**

### CODE REVIEW

File: firebase/test/core/domain/models/stored-public-collection.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import ValidationError from "#faults/errors/validation-error";
 import { storedCollectionSchema } from "#domain/models/collection";
 import { doesNotThrow, throws } from "assert";
 import { validate } from "#utils/validate";
-import { newRawLocalCollection } from "#test/core/data/schemas/collections-fakes";
+import { newRawStoredCollection } from "#test/core/data/schemas/collections-fakes";
```
Comment (ggirotto) [inline firebase/test/core/domain/models/stored-public-collection.spec.ts:6]: Also fixed `LocalCollection` usage instead `StoreLocation`

File: firebase/src/core/domain/models/collection.ts [hunk 2]
```diff
@@ -84,12 +84,12 @@ export const localCollectionSchema = publicCollectionSchema.append({
...
-  memosOrder: Joi.array().items(Joi.string()).min(1).required(),
-});
-// TODO: How to compare between same-object properties?
-// .assert(Joi.ref("memosAmount"), Joi.ref("memosOrder.length"));
+export const storedCollectionSchema = publicCollectionSchema
+  .append({
+    memosAmount: Joi.number().integer().min(1).required(),
+    memosOrder: Joi.array().items(Joi.string()).unique().min(1).required(),
+  })
+  .assert(".memosAmount", Joi.ref("memosOrder.length"));
```
Comment (matuella) [inline firebase/src/core/domain/models/collection.ts:92]: Oh that was tricky, really nice. Where did you find the notation for this usage? Maybe we could add a reference to the docs? I tried and didn't find anything.

**Review by matuella — APPROVED** [review]

File: firebase/src/core/domain/models/collection.ts (hunk 2 above)
Comment (ggirotto) [reply firebase/src/core/domain/models/collection.ts]: While searching where I found this solution, I found a better one here: https://github.com/sideway/joi/pull/1206.

Let me know what do you think

**Review by matuella — APPROVED** [review]

File: firebase/src/core/domain/models/collection.ts (hunk 2 above)
Comment (ggirotto) [reply firebase/src/core/domain/models/collection.ts]: Added docs

**Pull request merged**